### PR TITLE
[Mtg-464] prealloc

### PR DESCRIPTION
**Pull request by kstepanovdev**

### CODE REVIEW

File: programs/rewards/src/asserts.rs [hunk 1]
```diff
@@ -88,6 +72,19 @@ pub fn assert_pubkey_eq(given: &Pubkey, expected: &Pubkey) -> ProgramResult {
     }
 }
 
+pub fn assert_account_len(account: &AccountInfo, len: usize) -> ProgramResult {
+    if account.data_len() >= len {
```
Comment (StanChe) [inline programs/rewards/src/asserts.rs:76]: shouldn't we check for just equality here?

Comment (kstepanovdev) [reply programs/rewards/src/asserts.rs:76]: Why would it harm if an account is bigger for some reason?

Comment (StanChe) [reply programs/rewards/src/asserts.rs:76]: The only potential problem is if it's a different (bigger) account with the same owner, so we could spoil or misinterpret some data.

Comment (kstepanovdev) [reply programs/rewards/src/asserts.rs:76]: Hmm, but we read the precise number of bytes. Even if the passed account is 9999k long, we will read only `LEN`  bytes. It's not hard to fix, just wondering.

**Review by StanChe — APPROVED** [review]
👍

File: programs/rewards/src/state/mining.rs [hunk 2]
```diff
@@ -13,47 +16,43 @@ use solana_program::{
...
-use super::{AccountType, CumulativeIndex, WeightedStakeDiffs};
+use super::{AccountType, CumulativeIndex, MiningWeightedStakeDiffs};
 
 pub struct WrappedMining<'a> {
     pub mining: &'a mut Mining,
     /// This structures stores the weighted stake modifiers on the date,
     /// where staking ends. This modifier will be applied on the specified date to the global stake,
     /// so that rewards distribution will change. BTreeMap<unix_timestamp, modifier diff>
-    pub weighted_stake_diffs: &'a mut WeightedStakeDiffs,
+    pub weighted_stake_diffs: Rc<RefCell<&'a mut MiningWeightedStakeDiffs>>,
```
[the remark this answers is not in the document]
Comment (kstepanovdev) [reply programs/rewards/src/state/mining.rs:26]: Experiment leftover, I didn't notice that I pushed it. Thanks!

**Pull request merged**